### PR TITLE
BUG: Set success attribute in OptimizeResult. See #9801

### LINKED ISSUE #9801: dual_annealing does not set the success attribute in OptimizeResult
```
from scipy.optimize import dual_annealing, rosen
res = dual_annealing(rosen, [(-2, 2), (-2, 2)])
print(res.success)
```

Gives:

```
--------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/miniconda3/envs/dev3/lib/python3.7/site-packages/scipy/optimize/optimize.py in __getattr__(self, name)
    113         try:
--> 114             return self[name]
    115         except KeyError:

KeyError: 'success'

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
<ipython-input-4-402eff86d4dd> in <module>
      1 from scipy.optimize import dual_annealing, rosen
      2 res = dual_annealing(rosen, [(-2, 2), (-2, 2)])
----> 3 res.success

~/miniconda3/envs/dev3/lib/python3.7/site-packages/scipy/optimize/optimize.py in __getattr__(self, name)
    114             return self[name]
    115         except KeyError:
--> 116             raise AttributeError(name)
    117 
    118     __setattr__ = dict.__setitem__

AttributeError: success
```

### PR DESCRIPTION
**Pull request by sgubianpm**
closes gh-9801

### CODE REVIEW

File: scipy/optimize/_dual_annealing.py [hunk 1]
```diff
@@ -637,11 +637,16 @@ def dual_annealing(func, bounds, args=(), maxiter=1000,
     # Strategy chain instance
     strategy_chain = StrategyChain(accept, visit_dist, func_wrapper,
                                minimizer_wrapper, rand_state, energy_state)
-    # Run the search loop
     need_to_stop = False
     iteration = 0
     message = []
+    # OptimizeResult object to be returned
+    optimize_res = OptimizeResult()
+    optimize_res.sucess = True
```
Comment (rgommers) [inline scipy/optimize/_dual_annealing.py]: it's spelled with two `c`'s.

**Comment by rgommers** [comment]
Thanks @sgubianpm. Could you also add a regression test for this?

File: scipy/optimize/_dual_annealing.py (hunk 1 above)
Comment (sgubianpm) [reply scipy/optimize/_dual_annealing.py]: Fixed and added asserts in existing tests for this.

**Comment by rgommers** [comment]
LGTM now, merged. Thanks @sgubianpm

**Pull request merged**